File: cuda_tools/ops/src/nms.cpp
#include <torch/extension.h>
#include <vector>

using namespace at;

Tensor NMSCUDAKernelLauncher(Tensor boxes, float iou_threshold);

Tensor nms(Tensor boxes, float iou_threshold)
{
    return NMSCUDAKernelLauncher(boxes, iou_threshold);
}

File: cuda_tools/nms.py
import torch
import numpy as np
from cuda_tools import ops

class NMSop(torch.autograd.Function):

    @staticmethod
    def forward(ctx, bboxes, iou_threshold):
        inds = ops.nms(bboxes, iou_threshold=float(iou_threshold))

        return inds

def cuda_nms(boxes, iou_threshold):
    inds = NMSop.apply(boxes, iou_threshold)
    return inds